src/music/Envelopes.ts

```diff
@@ -0,0 +1,38 @@
+import { lerp } from "../util/Util.js";
+
+export interface AdsrConfig {
+    attack?: number;
+    decay?: number;
+    sustain?: number;
+    release?: number;
+}
+
+export namespace Envelopes {
+
+    export function createAdsrEnvelope(context: AudioContext, start: number, duration: number, env: AdsrConfig, volume: number = 1): GainNode {
+        // these are all optional, so make sure they're not undefined
+        const attack = env.attack || 0;
+        const decay = env.decay || 0;
+        const sustain = env.sustain || 0;
+        const release = env.release || 0;
```

Comment:
These can be a destructure with default values, e.g. `const { attack = 0, decay = 0, sustain = 0, release = 0 } = env;`